Kuka sim support package added depends on #12.

EXPX3 opened the pull request:
contains launch file for gazebo classic and gazebo simulators, base models independent of ros2_control support

Commit:
created new package  'kuka_ros2_control_support'

Commit:
single ros2_control launch file for all of the kuka robots  except 'kuka_lbr_iiwa_support'

Commit:
moved common_macro.ros2_control.xacro to kuka_ros2_control_support

Commit:
added new support package for simulators with ros2_control

Commit:
modifed to match other packages

Commit:
added common_macro_ros2_control_sim

Commit:
launch file for th classic

Commit:
adding inertial and modifying mesh files access path

Commit:
added common_sim_macro.ros2_control

Commit:
making base kuka files independent of ros2_control

Commit:
making base kuka files independent of ros2_control-2

Commit:
functional kuka simulators with ros2_control support

Commit:
functional ros2_control support for kuka base models

Commit:
removed ros2_control only support package

Commit:
added launch file for gazebo_sim

destogl commented on kuka_kr150_support/urdf/kr150r3100_2_macro.xacro:
```diff
@@ -1,118 +1,126 @@
 <?xml version="1.0"?>
 <robot xmlns:xacro="http://wiki.ros.org/xacro">
+  <xacro:include filename="$(find kuka_resources)/urdf/common_materials.xacro"/>
+  <xacro:include filename="$(find kuka_resources)/urdf/common_properties.xacro"/>
```
Those two files are also not needed here. They should be in the simulation xacro files

EXPX3 replied:
These were already here in some of the other base files, here only common_properties.xacro should have been added extra for getting gazebo to work, I can add it to simulation .xacro files this part as i added it for gazebo specific purpose. But the traces of the default_inertial tags will be there any way in the base files. If someone wants to use these base files alone without gazebo or ros2_control support, then they will get error as common_properties.xacro is not included in the base_macro that is using them instead in the simulation xacro.

 I am not sure if every other simulator out there needs inertial tags. some of the base files had inertial tags defined, i just added default_inertial wherever it was needed. I created this PR https://github.com/StoglRobotics-forks/kuka_experimental/pull/13 ,assuming default_inertial are needed in the base files. 
 
 i thin it should be included in the base_macro.xacro files instead of simulation .xacro files what do you think? @destogl

Robotgir replied:
check my comment here https://github.com/StoglRobotics-forks/kuka_experimental/pull/15#discussion_r1122972095
wait for your feedback

destogl commented on kuka_kr6_support/urdf/kr6r700sixx_macro.xacro:
```diff
@@ -1,111 +1,119 @@
 <?xml version="1.0"?>
 <robot xmlns:xacro="http://wiki.ros.org/xacro">
   <xacro:include filename="$(find kuka_resources)/urdf/common_constants.xacro"/>
```
do we still need common constants? If we are defining PI, then we should use `${pi}` instead.

EXPX3 replied:
common constants.xacro has only DEG2RAD, property defined and we are not using it, 
we are using ${radians()}, no need of constant.xacro

destogl replied:
That's great!

destogl commented on kuka_kr210_support/urdf/kr210l150_macro.xacro:
```diff
@@ -12,7 +12,7 @@
         <origin xyz="0 0 0" rpy="${-radians(90)} 0 0"/>
         <geometry>
-          <mesh filename="package://kuka_kr210_support/meshes/kr210l150/visual/base_link.dae"/>
+          <mesh filename="file://$(find kuka_kr210_support)/meshes/kr210l150/visual/base_link.dae"/>
```
Why is here no `default_intertial` used?

EXPX3 replied:
because it already had inertial tags defined

destogl commented on kuka_lbr_iiwa_support/urdf/lbr_iiwa_14_r820_macro.xacro:
```diff
@@ -3,114 +3,122 @@
   <xacro:macro name="kuka_lbr_iiwa_14_r820" params="prefix">
     <!-- link list -->
     <link name="${prefix}base_link">
+      <xacro:default_inertial/>
```
We should include file for default intertials here where we are also using it. Otherwise we have always add this in the file that is including this.

destogl commented on kuka_ros2_control_sim_support/launch/test_common_gazebo_classic_ros2_control.launch.py:
```diff
@@ -0,0 +1,298 @@
+        DeclareLaunchArgument(
+            "runtime_config_package",
+            default_value="kuka_resources",
+            description='Package with the controller\'s configuration in "config" folder. \
+        Usually the argument is not set, it enables use of a custom setup.',
```
Should probably be this to pass pre-commit

```suggestion
            description="Package with the controller's configuration in \"config\" folder. \
        Usually the argument is not set, it enables use of a custom setup.",
```

destogl commented on kuka_ros2_control_sim_support/launch/test_common_gazebo_classic_ros2_control.launch.py:
```diff
@@ -0,0 +1,298 @@
+            description="Robot controller to start.",
+        )
+    )
+    declared_arguments.append(
+        DeclareLaunchArgument(
+            "use_mock_hardware",
+            default_value="true",
+            description="Start robot with fake hardware mirroring command to its states.",
+        )
+    )
```
This file should be hard-coded for gazebo classic, so we don't need flag for mock hw

destogl commented on kuka_ros2_control_sim_support/launch/test_common_gazebo_classic_ros2_control.launch.py:
```diff
@@ -0,0 +1,298 @@
+            description="Start robot with fake hardware mirroring command to its states.",
+        )
+    )
+    declared_arguments.append(
+        DeclareLaunchArgument(
+            "ros2_control_sim_support_package",
+            default_value="kuka_ros2_control_sim_support",
+            description="Support package for ros2_control and simulators. Contains files that use base "
+            "kuka robot description and add on ros2_control support with gazebo simulators. ",
+        )
+    )
+    declared_arguments.append(
+        DeclareLaunchArgument(
+            "common_sim_description_file",
+            default_value="common_sim_kuka.xacro",
+            description="loads robot macros by chosen value for arguments 'robot_description_package'"
+            "'robot_description_file' and 'robot_name'. Also loads ros2_control with simulators support .",
+        )
+    )
```
I don't think this is needed, really. If someone wants to change this, they will probably have their launch file anyway.

destogl commented on kuka_ros2_control_sim_support/launch/test_common_gazebo_sim_ros2_control.launch.py:
```diff
@@ -0,0 +1,297 @@
+from launch_ros.actions import Node
+from launch_ros.substitutions import FindPackageShare
+
+def generate_launch_description():
```
apply the same comments as for gazebo classic file

destogl commented on kuka_ros2_control_sim_support/package.xml:
```diff
@@ -0,0 +1,29 @@
+
+  <buildtool_depend>ament_cmake</buildtool_depend>
+
+  <test_depend>roslaunch</test_depend>
```
This should not be dependency anymore in ROS2

destogl commented on kuka_ros2_control_sim_support/package.xml:
```diff
@@ -0,0 +1,29 @@
+
+  <test_depend>roslaunch</test_depend>
+
+  <exec_depend>industrial_robot_client</exec_depend>
```
This should not be needed here, and actually nowhere (maybe in the drivers)

```suggestion
```

destogl commented on kuka_ros2_control_sim_support/urdf/common_sim_kuka.xacro:
```diff
@@ -0,0 +1,93 @@
+    />
+    
+  <xacro:if value="${robot_description_package == 'kuka_kr3_support'}">
+    <xacro:kuka_kr3r540 prefix="$(arg prefix)"/>
```
not perfect, but I suspect that nothing else worked...

destogl replied:
Let's keep it like this. Maybe we add multiple files at some point.

EXPX3 replied:
> not perfect, but I suspect that nothing else worked...

yes tried ${}, as arguments call and as property call, nothing worked

destogl commented on kuka_ros2_control_sim_support/urdf/common_sim_macro.ros2_control.xacro:
```diff
@@ -1,14 +1,31 @@
 <robot xmlns:xacro="http://www.ros.org/wiki/xacro">
 
-  <xacro:macro name="kuka_common_ros2_control_macro" params="name prefix use_mock_hardware:=^|false">
+  <xacro:macro name="kuka_common_sim_ros2_control_macro" params="
```
I would put this into `*_ros2_control_*` package and sim package can use that one.

EXPX3 replied:
i put both files separately as we dont want traces of the gazebo in the  ros2_control only support package. they have their default values set in launch files, but still, for someone looking at ros2_control support package seeing gazebo related flags might be distracting, i thought @destogl

destogl reviewed (commented):
Except `sim_support` package we also need `ros2_control_support` package. So when we are using robot on the HW we don't need to install the simulators. This is dependency management thing.

Also, I have added some small comment to optimize files.

EXPX3 commented:
> Except `sim_support` package we also need `ros2_control_support` package. So when we are using robot on the HW we don't need to install the simulators. This is dependency management thing.
> 
> Also, I have added some small comment to optimize files.



> Except `sim_support` package we also need `ros2_control_support` package. So when we are using robot on the HW we don't need to install the simulators. This is dependency management thing.
> 
> Also, I have added some small comment to optimize files.

@destogl Below is the PR for ros2_control part alone
https://github.com/StoglRobotics-forks/kuka_experimental/pull/12

EXPX3 committed:
Update kuka_ros2_control_sim_support/package.xml

Co-authored-by: Dr. Denis <denis@stoglrobotics.de>

Commit:
removed unused xacro and added one useful

Commit:
removed mock_hardware flag

Commit:
removed arguments not necessary

EXPX3 committed:
Update kuka_ros2_control_sim_support/launch/test_common_gazebo_classic_ros2_control.launch.py

Co-authored-by: Dr. Denis <denis@stoglrobotics.de>

Commit:
removed dependency not needed in ros2

Commit:
ran pre-commit except *_rsi_*

Commit:
correction

Commit:
included kuka_lbr to run from common launch file

Commit:
ran-precommit

Commit:
added kuka_ros2_control_support along side kuka_ros2_control_sim_support

Commit:
making kuka_ros2_control_sim_support dependent on kuka_ros2_control_support

destogl commented on kuka_kr10_support/urdf/kr10r900_2_macro.xacro:
```diff
@@ -183,4 +191,4 @@
     </joint>
   </xacro:macro>
-</robot>
+</robot>
```
```suggestion
</robot>

```

destogl commented on kuka_kr5_support/urdf/kr5_arc_macro.xacro:
```diff
@@ -284,4 +284,4 @@
 
   </xacro:macro>
-</robot>
+</robot>
```
```suggestion
</robot>

```

destogl commented on kuka_ros2_control_sim_support/package.xml:
```diff
@@ -0,0 +1,26 @@
+  <name>kuka_ros2_control_sim_support</name>
+  <version>0.0.0</version>
+  <description>ros2_control simulator, Gazebo-classic and Gazebo support package</description>
+  <maintainer email="giridhar6937@gmail.com">Giridhar Bukka</maintainer>
```
```suggestion
  <maintainer email="denis.stogl@stoglrobotics.de">Dr.-Ing. Denis Štogl</maintainer>
  <maintainer email="giridhar.buka@stoglrobotics.de">Giridhar Bukka</maintainer>
```

destogl commented on kuka_ros2_control_sim_support/launch/test_common_gazebo_sim_ros2_control.launch.py:
```diff
@@ -0,0 +1,270 @@
+
+    # Spawn robot
+    ignition_spawn_robot = Node(
+        package="ros_ign_gazebo",
```
Please use instead:

```suggestion
        package="ros_gz_sim",
```

destogl commented on kuka_ros2_control_sim_support/launch/test_common_gazebo_sim_ros2_control.launch.py:
```diff
@@ -0,0 +1,270 @@
+    )
+
+    # # Gazebo nodes
+    gazebo_ign = IncludeLaunchDescription(
```
```suggestion
    gazebo_sim = IncludeLaunchDescription(
```

destogl commented on kuka_ros2_control_sim_support/launch/test_common_gazebo_sim_ros2_control.launch.py:
```diff
@@ -0,0 +1,270 @@
+    )
+
+    # Spawn robot
+    ignition_spawn_robot = Node(
```
```suggestion
    gazebo_spawn_robot = Node(
```

destogl commented on kuka_ros2_control_sim_support/launch/test_common_gazebo_sim_ros2_control.launch.py:
```diff
@@ -0,0 +1,270 @@
+    return LaunchDescription(
+        declared_arguments
+        + [
+            gazebo_ign,
```
```suggestion
            gazebo_sim,
```

destogl commented on kuka_ros2_control_sim_support/launch/test_common_gazebo_sim_ros2_control.launch.py:
```diff
@@ -0,0 +1,270 @@
+        declared_arguments
+        + [
+            gazebo_ign,
+            ignition_spawn_robot,
```
```suggestion
            gazebo_spawn_robot,
```

destogl commented on kuka_ros2_control_sim_support/launch/test_common_gazebo_sim_ros2_control.launch.py:
```diff
@@ -0,0 +1,270 @@
+    # # Gazebo nodes
+    gazebo_ign = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            [FindPackageShare("ros_ign_gazebo"), "/launch", "/ign_gazebo.launch.py"]
```
```suggestion
            [FindPackageShare("ros_gz_sim"), "/launch", "/gz_sim.launch.py"]
```

destogl commented on kuka_ros2_control_sim_support/launch/test_common_gazebo_classic_ros2_control.launch.py:
```diff
@@ -0,0 +1,275 @@
+    )
+
+    # # Gazebo node
+    gazebo_classic = IncludeLaunchDescription(
```
```suggestion
    gazebo = IncludeLaunchDescription(
```

destogl commented on kuka_ros2_control_sim_support/launch/test_common_gazebo_classic_ros2_control.launch.py:
```diff
@@ -0,0 +1,275 @@
+
+    # Spawn robot
+
+    classic_spawn_robot = Node(
```
```suggestion
    gazebo_spawn_robot = Node(
```

destogl commented on kuka_ros2_control_sim_support/launch/test_common_gazebo_classic_ros2_control.launch.py:
```diff
@@ -0,0 +1,275 @@
+    return LaunchDescription(
+        declared_arguments
+        + [
+            gazebo_classic,
+            classic_spawn_robot,
```
```suggestion
            gazebo,
            gazebo_spawn_robot,
```

destogl commented on kuka_ros2_control_sim_support/package.xml:
```diff
@@ -0,0 +1,26 @@
+
+  <buildtool_depend>ament_cmake</buildtool_depend>
+
+  <exec_depend>joint_state_publisher</exec_depend>
+  <exec_depend>joint_state_broadcaster</exec_depend>
+  <exec_depend>robot_state_publisher</exec_depend>
+  <exec_depend>joint_state_publisher_gui</exec_depend>
+  <exec_depend>controller_manager</exec_depend>
+  <exec_depend>forward_command_controller</exec_depend>
+  <exec_depend>ros2_control</exec_depend>
+  <exec_depend>ros2_controllers</exec_depend>
+  <exec_depend>rviz</exec_depend>
+  <exec_depend>xacro</exec_depend>
```
```suggestion
  <exec_depend>controller_manager</exec_depend>
  <exec_depend>forward_command_controller</exec_depend>
  <exec_depend>gazebo_ros2_control</exec_depend>
  <exec_depend>gz_ros2_control</exec_depend>
  <exec_depend>joint_state_publisher</exec_depend>
  <exec_depend>joint_state_broadcaster</exec_depend>
  <exec_depend>robot_state_publisher</exec_depend>
  <exec_depend>joint_state_publisher_gui</exec_depend>
  <exec_depend>ros_gz_sim</exec_depend>
  <exec_depend>kuka_ros2_control_support</exec_depend>
  <exec_depend>rviz</exec_depend>
  <exec_depend>xacro</exec_depend>
```

destogl commented on kuka_ros2_control_support/CMakeLists.txt:
```diff
@@ -0,0 +1,19 @@
+  DIRECTORY urdf launch
+  DESTINATION share/${PROJECT_NAME}
+)
+
```
```suggestion
```

destogl commented on kuka_ros2_control_support/package.xml:
```diff
@@ -0,0 +1,28 @@
+  <name>kuka_ros2_control_support</name>
+  <version>0.0.0</version>
+  <description>ros2_control support package for kuka</description>
+  <maintainer email="giridhar6937@gmail.com">Giridhar Bukka</maintainer>
```
```suggestion
  <maintainer email="denis.stogl@stoglrobotics.de">Dr.-Ing. Denis Štogl</maintainer>
  <maintainer email="giridhar.bukka@stoglrobotics.de">Giridhar Bukka</maintainer>
```

destogl commented:
Please also add manual into README.md on how this repository can be used, dependencies installed and robot, mock hardware and simulation can be started.

destogl committed:
Merge branch 'rolling' into kuka_ros2_control_sim_support-package-added

EXPX3 committed:
Update kuka_ros2_control_sim_support/package.xml

Co-authored-by: Dr. Denis <denis@stoglrobotics.de>

EXPX3 committed:
Update kuka_ros2_control_sim_support/launch/test_common_gazebo_sim_ros2_control.launch.py

Co-authored-by: Dr. Denis <denis@stoglrobotics.de>

EXPX3 committed:
Update kuka_ros2_control_support/package.xml

Co-authored-by: Dr. Denis <denis@stoglrobotics.de>

EXPX3 committed:
Apply suggestions from code review

Co-authored-by: Dr. Denis <denis@stoglrobotics.de>

EXPX3 commented on kuka_ros2_control_support/package.xml:
```diff
@@ -0,0 +1,29 @@
+
+  <test_depend>roslaunch</test_depend>
+
+  <exec_depend>joint_state_publisher</exec_depend>
+  <exec_depend>joint_state_broadcaster</exec_depend>
+  <exec_depend>robot_state_publisher</exec_depend>
+  <exec_depend>joint_state_publisher_gui</exec_depend>
+  <exec_depend>controller_manager</exec_depend>
+  <exec_depend>forward_command_controller</exec_depend>
+  <exec_depend>ros2_control</exec_depend>
+  <exec_depend>ros2_controllers</exec_depend>
+  <exec_depend>rviz</exec_depend>
+  <exec_depend>xacro</exec_depend>
```
```suggestion
  <exec_depend>controller_manager</exec_depend>
  <exec_depend>forward_command_controller</exec_depend>
  <exec_depend>joint_state_publisher</exec_depend>
  <exec_depend>joint_state_broadcaster</exec_depend>
  <exec_depend>joint_state_publisher_gui</exec_depend>
  <exec_depend>robot_state_publisher</exec_depend>
  <exec_depend>ros2_control</exec_depend>
  <exec_depend>ros2_controllers</exec_depend>
  <exec_depend>rviz</exec_depend>
  <exec_depend>xacro</exec_depend>
```

EXPX3 committed:
Apply suggestions from code review

Co-authored-by: Dr. Denis <denis@stoglrobotics.de>

Commit:
renaming

Commit:
correction

Commit:
added dependency

Commit:
added dependency

Commit:
modified readme

Commit:
update and upgrade

EXPX3 commented on kuka_ros2_control_sim_support/package.xml:
```diff
@@ -0,0 +1,30 @@
+  <exec_depend>ros_gz_sim</exec_depend>
+  <exec_depend>kuka_ros2_control_support</exec_depend>
+  <exec_depend>rviz</exec_depend>
+  <exec_depend>xacro</exec_depend>ign-ros2-control
```
```suggestion
  <exec_depend>xacro</exec_depend>
```

EXPX3 committed:
Update kuka_ros2_control_sim_support/package.xml

Commit:
clarity

muritane commented on kuka_ros2_control_sim_support/launch/test_common_gazebo_classic_ros2_control.launch.py:
```diff
@@ -0,0 +1,275 @@
+        arguments=["joint_state_broadcaster", "--controller-manager", "/controller_manager"],
+    )
+
+    robot_controllers = [robot_controller]
+    robot_controller_spawners = []
+    for controller in robot_controllers:
```
robot_controllers variable already exists above, should this one be changed?
```suggestion
    robot_controllers_to_spawn = [robot_controller]
    robot_controller_spawners = []
    for controller in robot_controllers_to_spawn:
```

muritane commented on kuka_ros2_control_sim_support/launch/test_common_gazebo_classic_ros2_control.launch.py:
```diff
@@ -0,0 +1,275 @@
+        DeclareLaunchArgument(
+            "robot_controller",
+            default_value="forward_position_controller",
+            choices=["forward_position_controller", "joint_trajectory_controller"],
```
there is no joint_trajectory_controller, but position_trajectory_controller, rename this one or in the configs
```suggestion
            choices=["forward_position_controller", "position_trajectory_controller"],
```

destogl committed:
Update kuka_ros2_control_sim_support/launch/test_common_gazebo_classic_ros2_control.launch.py

Co-authored-by: muritane <31107191+muritane@users.noreply.github.com>

destogl committed:
Update kuka_ros2_control_sim_support/launch/test_common_gazebo_classic_ros2_control.launch.py

Co-authored-by: muritane <31107191+muritane@users.noreply.github.com>

destogl commented:
@muritane you had script to test this PR. Can we upload this for now?

muritane committed:
added test for kuka_ros2_control_sim_support
